# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/painting/nsDisplayList.h
# Commit: 33dbc637a003
# Full Hash: 33dbc637a003ff6431860440325a5ece32a1c91d
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2021-05-19 08:32:22
# Regressor Bug: 1540737
# File Overlap Count: 2
# Description:
#   Bug 1540737 - Allow filter painting to take a callback for painting children. r=miko
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D114670
# ==============================================================================

diff -r d7a58cff3e92 -r 33dbc637a003 layout/painting/nsDisplayList.h
--- a/layout/painting/nsDisplayList.h	Wed May 19 00:38:57 2021 +0000
+++ b/layout/painting/nsDisplayList.h	Wed May 19 01:11:30 2021 +0000
@@ -6649,6 +6649,10 @@
   void PaintAsLayer(nsDisplayListBuilder* aBuilder, gfxContext* aCtx,
                     LayerManager* aManager);
 
+  void PaintWithContentsPaintCallback(
+      nsDisplayListBuilder* aBuilder, gfxContext* aCtx,
+      const std::function<void(gfxContext* aContext)>& aPaintChildren);
+
   bool CreateWebRenderCommands(
       mozilla::wr::DisplayListBuilder& aBuilder,
       mozilla::wr::IpcResourceUpdateQueue& aResources,